### ISSUE TITLE
[Bug]: KNN build expects "lib" for older versions

### ISSUE THREAD

**Issue by jmazanec15** [opensearch-project/opensearch-build#2082]
### Describe the bug

As a part of https://github.com/opensearch-project/k-NN/issues/314, we switched the location of the k-NN libraries from knnlib to lib. The changes on the plugin side only took effect for 2.x, because the campaign was for 2.0. 

This PR (https://github.com/opensearch-project/opensearch-build/pull/1866) updated the build scripts to expect the libs in "lib" not "knnlib" in this repo. 

However, because the build repo is not tied to a version of OpenSearch, the 1.3.2 builds expected the libraries to be located in lib instead of knnlib.

In terms of solutions, it seems like either we could (1) tie the build components to a version or (2) insert if/else statements based on version somewhere in the scripts, or (3) backport the knnlib->lib name change to 1.x and 1.3 in the plugin. (3) sounds the most practical to me, but it seems like (1) would make sense for cases like this in the long run.

### To reproduce

See https://github.com/opensearch-project/k-NN/issues/381

### Expected behavior

```
>= 2.0 builds should expect libraries in `lib` directory.
< 2.0 builds should expect libraries in `knnlib` directory.
```

### Screenshots

N/A

### Host / Environment

N/A

### Additional context

N/A

### Relevant log output

```shell
java.lang.UnsatisfiedLinkError: no opensearchknn_nmslib in java.library.path: [., /home/test/opensearch-1.3.2/plugins/opensearch-knn/lib, /usr/java/packages/lib, /usr/lib64, /lib64, /lib, /usr/lib]
        at java.base/java.lang.ClassLoader.loadLibrary(ClassLoader.java:2680)
        at java.base/java.lang.Runtime.loadLibrary0(Runtime.java:807)
        at java.base/java.lang.System.loadLibrary(System.java:1907)
        at org.opensearch.knn.jni.NmslibService.lambda$static$0(NmslibService.java:34)
        at java.base/java.security.AccessController.doPrivileged(AccessController.java:312)
        at org.opensearch.knn.jni.NmslibService.<clinit>(NmslibService.java:33)
        at org.opensearch.knn.jni.JNIService.createIndex(JNIService.java:36)
        at org.opensearch.knn.index.codec.KNN80Codec.KNN80DocValuesConsumer.lambda$createKNNIndexFromScratch$1(KNN80DocValuesConsumer.java:198)
        at java.base/java.security.AccessController.doPrivileged(AccessController.java:312)
        at org.opensearch.knn.index.codec.KNN80Codec.KNN80DocValuesConsumer.createKNNIndexFromScratch(KNN80DocValuesConsumer.java:196)
        at org.opensearch.knn.index.codec.KNN80Codec.KNN80DocValuesConsumer.addKNNBinaryField(KNN80DocValuesConsumer.java:122)
        at org.opensearch.knn.index.codec.KNN80Codec.KNN80DocValuesConsumer.addBinaryField(KNN80DocValuesConsumer.java:70)
        at org.apache.lucene.codecs.perfield.PerFieldDocValuesFormat$FieldsWriter.addBinaryField(PerFieldDocValuesFormat.java:114)
        at org.apache.lucene.index.BinaryDocValuesWriter.flush(BinaryDocValuesWriter.java:123)
        at org.apache.lucene.index.DefaultIndexingChain.writeDocValues(DefaultIndexingChain.java:348)
        at org.apache.lucene.index.DefaultIndexingChain.flush(DefaultIndexingChain.java:228)
        at org.apache.lucene.index.DocumentsWriterPerThread.flush(DocumentsWriterPerThread.java:350)
        at org.apache.lucene.index.DocumentsWriter.doFlush(DocumentsWriter.java:476)
        at org.apache.lucene.index.DocumentsWriter.flushAllThreads(DocumentsWriter.java:656)
        at org.apache.lucene.index.IndexWriter.getReader(IndexWriter.java:605)
        at org.apache.lucene.index.StandardDirectoryReader.doOpenFromWriter(StandardDirectoryReader.java:293)
        at org.apache.lucene.index.StandardDirectoryReader.doOpenIfChanged(StandardDirectoryReader.java:268)
        at org.apache.lucene.index.StandardDirectoryReader.doOpenIfChanged(StandardDirectoryReader.java:258)
        at org.apache.lucene.index.FilterDirectoryReader.doOpenIfChanged(FilterDirectoryReader.java:112)
        at org.apache.lucene.index.DirectoryReader.openIfChanged(DirectoryReader.java:173)
        at org.opensearch.index.engine.OpenSearchReaderManager.refreshIfNeeded(OpenSearchReaderManager.java:81)
        at org.opensearch.index.engine.OpenSearchReaderManager.refreshIfNeeded(OpenSearchReaderManager.java:53)
        at org.apache.lucene.search.ReferenceManager.doMaybeRefresh(ReferenceManager.java:176)
        at org.apache.lucene.search.ReferenceManager.maybeRefreshBlocking(ReferenceManager.java:253)
        at org.opensearch.index.engine.InternalEngine$ExternalReaderManager.refreshIfNeeded(InternalEngine.java:400)
        at org.opensearch.index.engine.InternalEngine$ExternalReaderManager.refreshIfNeeded(InternalEngine.java:380)
        at org.apache.lucene.search.ReferenceManager.doMaybeRefresh(ReferenceManager.java:176)
        at org.apache.lucene.search.ReferenceManager.maybeRefreshBlocking(ReferenceManager.java:253)
        at org.opensearch.index.engine.InternalEngine.refresh(InternalEngine.java:1897)
        at org.opensearch.index.engine.InternalEngine.refresh(InternalEngine.java:1874)
        at org.opensearch.index.shard.IndexShard.refresh(IndexShard.java:1201)
        at org.opensearch.action.support.replication.TransportWriteAction$AsyncAfterWriteAction.<init>(TransportWriteAction.java:420)
        at org.opensearch.action.support.replication.TransportWriteAction$WritePrimaryResult.runPostReplicationActions(TransportWriteAction.java:305)
        at org.opensearch.action.support.replication.ReplicationOperation.handlePrimaryResult(ReplicationOperation.java:176)
        at org.opensearch.action.ActionListener$1.onResponse(ActionListener.java:78)
        at org.opensearch.action.ActionListener.completeWith(ActionListener.java:351)
        at org.opensearch.action.bulk.TransportShardBulkAction$2.finishRequest(TransportShardBulkAction.java:264)
        at org.opensearch.action.bulk.TransportShardBulkAction$2.doRun(TransportShardBulkAction.java:227)
        at org.opensearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:50)
        at org.opensearch.action.bulk.TransportShardBulkAction.performOnPrimary(TransportShardBulkAction.java:276)
        at org.opensearch.action.bulk.TransportShardBulkAction.dispatchedShardOperationOnPrimary(TransportShardBulkAction.java:165)
        at org.opensearch.action.bulk.TransportShardBulkAction.dispatchedShardOperationOnPrimary(TransportShardBulkAction.java:98)
        at org.opensearch.action.support.replication.TransportWriteAction$1.doRun(TransportWriteAction.java:220)
        at org.opensearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:792)
        at org.opensearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:50)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
        at java.base/java.lang.Thread.run(Thread.java:832)
```
```

**Comment by peterzhuamazon** [comment]
We can use 1.3.1 tag in build repo to build.

**Comment by peterzhuamazon** [comment]
https://github.com/opensearch-project/opensearch-build/releases/tag/1.3.1

**Comment by jmazanec15** [comment]
I should be more specific - build scripts are not causing issue. It is the install/run time scripts (Dockerfile as well as tar install)

**Comment by peterzhuamazon** [comment]
@zelinh seems like this is not an issue anymore since we have shipped 1.3.2, please close.
Thanks.

**Comment by bbarani** [comment]
I am closing this issue as we have back ported the changes to support 1.x version. Please feel free to re-open if needed.
